### PR TITLE
MYB 149

### PR DESCRIPTION
**Pull request by julietapoume**
Handle errors during existing interactions with user, and display a notification alert indicating that an error occurred:
- App could't locate current user location
- No results for search
- A field is empty and it's required

myb149

### CODE REVIEW

File: MyBus/MyBus/SearchViewController.swift [hunk 1]
```diff
@@ -92,65 +96,123 @@ class SearchViewController: UIViewController, UITableViewDelegate
...
+                            self.getBusLines(latitudeOrigin, longitudeOrigin: longitudeOrigin, latDestination: latitudeDestination, lngDestination: longitudeDestination)
+                        default:
+                            self.generateAlert("No sabemos donde es el destino", message: "No pudimos resolver la dirección de destino ingresada")
+                            break
+                        }
+                    }
+                default:
+                    self.generateAlert("No sabemos donde es el destino", message: "No pudimos resolver la dirección de destino ingresada")
+                    break
+                }
```
Comment (pablotdl) [inline MyBus/MyBus/SearchViewController.swift:181]: Can't you reuse some of the code and avoid duplication?

Comment (julietapoume) [inline MyBus/MyBus/SearchViewController.swift:181]: Yes, but is better resolve this in other task.

**Comment by lisandro** [comment]
Hi @julietapoume code seems to be good. To prevent Swift style diffs in changes, please install swiftlint `brew install swiftlint` and run `swiftlint autocorrect` from root of project where we have linting rules. :metal:

File: MyBus/MyBus/MainViewController.swift [hunk 2]
```diff
@@ -96,8 +96,18 @@ class MainViewController: UIViewController, MapBusRoadDelegate {
             self.cycleViewController(self.currentViewController!, toViewController: self.mapViewController)
             self.currentViewController = self.mapViewController
         }
-        self.mapViewController.mapView.showsUserLocation = true
-        self.mapViewController.mapView.setZoomLevel(16, animated: false)
+        //check the status of localization        
+        if (self.mapViewController.mapView.userLocation != nil)
```
Comment (lisandro) [inline MyBus/MyBus/MainViewController.swift]: It appears that does not work as expected. I tested turning on location services having `userLocation == nil` and tapping in refresh button userLocation is not updated or tried to locate. Please take a look this [delegate method](https://www.mapbox.com/ios-sdk/api/3.3.3/Protocols/MGLMapViewDelegate.html#/c:objc%28pl%29MGLMapViewDelegate%28im%29mapView:didFailToLocateUserWithError:) where we can handle this exception

Comment (julietapoume) [inline MyBus/MyBus/MainViewController.swift]: @lisandro the method was modified.

**Comment by safink** [comment]
Approved

File: MyBus/MyBus/MainViewController.swift (hunk 2 above)
Comment (lisandro) [inline MyBus/MyBus/MainViewController.swift]: @julietapoume seems better. To consider case of location services disabled for all apps I suggest replace with this code.

```
let locationServiceAuth = CLLocationManager.authorizationStatus()
guard (locationServiceAuth == .AuthorizedAlways || locationServiceAuth == .AuthorizedWhenInUse) else {
    let alert = UIAlertController.init(title: "No te encontramos en el mapa", message: "Si queres que te ubiquemos en el mapa activa tu localizacion.", preferredStyle: .Alert)
    let action = UIAlertAction.init(title: "OK", style: UIAlertActionStyle.Default, handler: nil)
    alert.addAction(action)
    return self.presentViewController(alert, animated: true, completion: nil)
}

self.mapViewController.mapView.showsUserLocation = true
self.mapViewController.mapView.setZoomLevel(16, animated: false)
```

**Comment by lisandro** [comment]
Changes are OK and ready to be merged :+1:

**Pull request merged**